internlm/initialize/launch.py

```diff
@@ -474,3 +475,60 @@ def get_config_value(config, key, defalut):
     except KeyError:
         value = defalut
     return value
+
+
+# check pacakge
+try:
+    import numa
+    from numa import memory, schedule
+    from pynvml.smi import nvidia_smi
+except (AttributeError, ImportError):
+    get_numa = False
+    global_rank = int(os.environ["SLURM_PROCID"]) if "SLURM_PROCID" in os.environ else int(os.environ["RANK"])
+    if global_rank == 0:
+        logger.info(
+            "Try bind numa failed! Package import error, if numa is not installed, "
+            "please implement: pip install --upgrade py-libnuma"
+        )
+else:
+    get_numa = True
+
+
+def try_bind_numa(launcher):
+    # Early return if numa module not available
+    if not get_numa:
```

Comment:
Moving the code related to importing numa package into 'try_bind_numa' might make it easier to maintain.